src/celllistimpl.h

```diff
@@ -0,0 +1,911 @@
+
+  public:
+    using AbstractCellList =
+        typename AbstractSortableCellList<ContainerTypeOf<TContainer>, GridTypeOf<TGrid>>::AbstractCellList;
+    using Grid = TGrid;
+    using Container = TContainer;
+    using typename AbstractCellList::CellCoord; //!< grid (cell) coordinates type
+    using typename AbstractCellList::CellIndex; //!< grid (cell) index type
+    using typename AbstractCellList::Member;    //!< member type
+    using typename AbstractCellList::Members;   //!< members type
+
+    /**
+     * @brief Get members in a sorted order.
+     * @param cell_coordinates
+     * @return  sorted members
+     */
+    const Members& getSortedMembers(const CellCoord& cell_coordinates) override {
+        return getSorted(this->index(cell_coordinates));
+    }
+
+    /**
+     * @brief Get members of a neighbor cell in a sorted order.
+     * @param cell_coordinates
+     * @param cell_offset
+     * @return  sorted members
+     */
+    const Members& getNeighborSortedMembers(const CellCoord& cell_coordinates, const CellCoord& cell_offset) override {
+        if (this->isNeighborCell(cell_coordinates, cell_offset)) {
+            return getSortedMembers(cell_coordinates + cell_offset);
+        } else {
+            return this->getEmpty();
+        }
+    }
+
+    /**
+     *  @brief Construct an empty cell list from a grid.
+     *  @param cell_grid
+     */
+    SortableCellList(const GridOf<TBase>& cell_grid) : TBase(cell_grid) { is_sorted.resize(this->size(), false); }
+
+    /**
+     *  @brief Construct an empty cell list knowing a box dimension and minimal cell dimension.
+     *  @param box  spatial dimension of the box
+     *  @param cell_dimension  minimal length of the cell's edge
+     */
+    SortableCellList(const typename GridOf<TBase>::Point& box, const typename GridOf<TBase>::SpaceAxis cell_dimension)
+        : TBase(box, cell_dimension) {
+        is_sorted.resize(this->size(), false);
+    }
+
+  protected:
+    void add(const Member& member, const CellIndex& new_cell_index) {
+        TBase::add(member, new_cell_index);
+        if (is_sorted[new_cell_index]) {
+            is_sorted[new_cell_index] = false;
+        }
+    }
+
+    void remove(const Member& member, const CellIndex& old_cell_index) {
+        TBase::remove(member, old_cell_index);
+        if (is_sorted[old_cell_index]) {
+            is_sorted[old_cell_index] = false;
+        }
+    }
+
+    /**
+     * @brief Get members in a sorted order.
+     * @param cell_index
+     * @return  sorted members
+     */
+    const Members& getSorted(const CellIndex cell_index) {
+        auto& members = this->get(cell_index);
+        if (!is_sorted[cell_index]) {
+            std::sort(members.begin(), members.end());
+            is_sorted[cell_index] = true;
+        }
+        assert(std::is_sorted(members.begin(), members.end()));
+        return members;
+    }
+
+  private:
+    // TODO perhaps a map for sparse containers
```

Comment:
I think vector is fine; the STL implementation is supposedly using optimized bit storage for std::vector<bool>